xchange-bittrex/src/main/java/org/knowm/xchange/bittrex/service/BittrexTradeService.java

```diff
@@ -106,11 +125,29 @@ public OpenOrdersParams createOpenOrdersParams() {
     try {
       List<Order> orders = new ArrayList<>();
       for (String orderId : orderIds) {
-        BittrexOrder order = getBittrexOrder(orderId);
-        if (order != null) {
-          LimitOrder limitOrder = BittrexAdapters.adaptOrder(order);
-          orders.add(limitOrder);
+        BittrexOrder bittrexOrder = getBittrexOrder(orderId);
+        if (bittrexOrder == null) {
+          continue;
+        }
+        Order order = BittrexAdapters.adaptOrder(bittrexOrder);
+        if (order instanceof MarketOrder) {
+          BigDecimal rate =
+              decorateApiCall(
+                      () -> {
+                        List<BittrexExecution> executions =
+                            getBittrexOrderExecutions(order.getId());
+                        return executions.stream()
+                            .map(BittrexExecution::getRate)
+                            .reduce(BigDecimal.ZERO, BigDecimal::add)
+                            .divide(
+                                new BigDecimal(executions.size()),
+                                new MathContext(8, RoundingMode.FLOOR));
+                      })
+                  .withRetry(retry("orderExecutionRate"))
+                  .call();
+          order.setAveragePrice(rate);
```

Comment:
Sorry for being pedantic, trying to keep the code very readable here, do you think you could do something like 

```
          List<BittrexExecution> executions = 
              decorateApiCall(() -> getBittrexOrderExecutions(order.getId()))
              .withRetry(retry("orderExecutionRate"))
              .call();
          BigDecimal rate = executions.stream()
              .map(BittrexExecution::getRate)
              .reduce(BigDecimal.ZERO, BigDecimal::add)
              .divide(
                      new BigDecimal(executions.size()),
                      new MathContext(8, RoundingMode.FLOOR));
          order.setAveragePrice(rate);
```
To help with separating what is the actual API call vs Xchange business logic